pkg/controllers/osc/osc_controller.go

```diff
@@ -313,9 +327,51 @@ func (r *Reconciler) deleteGeneratedSecrets(ctx context.Context, md *clusterv1al
 	return nil
 }
 
+func (r *Reconciler) handleOSCAndSecretRotation(ctx context.Context, md *clusterv1alpha1.MachineDeployment) error {
+	oscName := fmt.Sprintf(resources.MachineDeploymentSubresourceNamePattern, md.Name, md.Namespace, resources.ProvisioningCloudConfig)
+	osc := &osmv1alpha1.OperatingSystemConfig{}
+	if err := r.Get(ctx, types.NamespacedName{Name: oscName, Namespace: r.namespace}, osc); err != nil && !kerrors.IsNotFound(err) {
+		return err
+	} else if err != nil && kerrors.IsNotFound(err) {
```

Comment:
Why this is needed? aren't you handling the error already in the line 333 ?

Reply:
Different conditions
1. Check if there is an error, other than `Resource not found`, then return error.
2. Check if there is an error, the error is `Resource not found`, then just return nil.

Reply:
Updated.